controllers/handlers/passt/passt.go

```diff
@@ -0,0 +1,334 @@
+	"os"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	netattdefv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	securityv1 "github.com/openshift/api/security/v1"
+
+	kubevirtcorev1 "kubevirt.io/api/core/v1"
+
+	hcov1beta1 "github.com/kubevirt/hyperconverged-cluster-operator/api/v1beta1"
+	"github.com/kubevirt/hyperconverged-cluster-operator/controllers/operands"
+	"github.com/kubevirt/hyperconverged-cluster-operator/pkg/nodeinfo"
+	hcoutil "github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
+)
+
+const (
+	DeployPasstNetworkBindingAnnotation = "deployPasstNetworkBinding"
+
+	BindingName = "passt"
+
+	passtCNIObjectName = "passt-binding-cni"
+
+	networkBindingNADName       = "primary-udn-kubevirt-binding"
+	networkBindingNADNamespace  = "default"
+	NetworkAttachmentDefinition = networkBindingNADNamespace + "/" + networkBindingNADName
+
+	bindingComputeMemoryOverhead = "250Mi"
+)
+
+var passtResourceMemory = resource.MustParse(bindingComputeMemoryOverhead)
+
+// CheckPasstImagesEnvExists checks if the passt image environment variable exists
+func CheckPasstImagesEnvExists() error {
+	if _, passtImageVarExists := os.LookupEnv(hcoutil.PasstImageEnvV); !passtImageVarExists {
+		return fmt.Errorf("the %s environment variable must be set", hcoutil.PasstImageEnvV)
+	}
+	if _, passtCNIImageVarExists := os.LookupEnv(hcoutil.PasstCNIImageEnvV); !passtCNIImageVarExists {
+		return fmt.Errorf("the %s environment variable must be set", hcoutil.PasstCNIImageEnvV)
+	}
+	return nil
+}
+
+// NetworkBinding creates an InterfaceBindingPlugin for passt network binding
+func NetworkBinding() kubevirtcorev1.InterfaceBindingPlugin {
+	return kubevirtcorev1.InterfaceBindingPlugin{
+		NetworkAttachmentDefinition: NetworkAttachmentDefinition,
+		SidecarImage:                os.Getenv(hcoutil.PasstImageEnvV),
+		Migration:                   &kubevirtcorev1.InterfaceBindingMigration{},
+		ComputeResourceOverhead: &kubevirtcorev1.ResourceRequirementsWithoutClaims{
+			Requests: corev1.ResourceList{
+				corev1.ResourceMemory: passtResourceMemory,
+			},
+		},
+	}
+}
+
+// NewPasstBindingCNISA creates a ServiceAccount for the passt binding CNI
+func NewPasstBindingCNISA(hc *hcov1beta1.HyperConverged) *corev1.ServiceAccount {
+	return &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      passtCNIObjectName,
+			Namespace: hc.Namespace,
+			Labels:    hcoutil.GetLabels(hcoutil.HyperConvergedName, hcoutil.AppComponentNetwork),
+		},
+	}
+}
+
+// NewPasstBindingCNIDaemonSet creates a DaemonSet for the passt binding CNI
+func NewPasstBindingCNIDaemonSet(hc *hcov1beta1.HyperConverged) *appsv1.DaemonSet {
+	maxUnavailable := intstr.FromString("10%")
```

Comment:
This is a compute of a constant value. can be done once, by defining the variable outside of the function.

Reply:
happens once (via cache)